sdks/python/apache_beam/typehints/native_type_compatibility_test.py

```diff
@@ -103,6 +98,64 @@ def test_convert_to_beam_types(self):
         native_type_compatibility.convert_to_beam_types(typing_types),
         beam_types)
 
+  def test_is_sub_class(self):
+    self.assertTrue(native_type_compatibility._safe_issubclass(
+        parent=typing.Dict,
+        derived=typing.Dict[bytes, int]))
+    self.assertFalse(native_type_compatibility._safe_issubclass(
+        parent=typing.List,
+        derived=typing.Dict[bytes, int]))
+    self.assertTrue(native_type_compatibility._safe_issubclass(
+        parent=typing.List,
+        derived=typing.List[bytes]))
+    self.assertFalse(native_type_compatibility._safe_issubclass(
+        parent=typing.List,
+        derived=typing.Dict[bytes, int]))
+    self.assertTrue(native_type_compatibility._safe_issubclass(
+        parent=typing.Set,
+        derived=typing.Set[int]))
+    self.assertFalse(native_type_compatibility._safe_issubclass(
+        parent=typing.List,
+        derived=typing.Set[float]))
+    self.assertTrue(native_type_compatibility._safe_issubclass(
+        parent=typing.Tuple,
+        derived=typing.Tuple[int]))
+    self.assertFalse(native_type_compatibility._safe_issubclass(
+        parent=typing.List,
+        derived=typing.Tuple[bytes]))
+
+  @unittest.skipIf(sys.version_info >= (2, 7, 0),
+                   'Order dosent matter in python 3')
+  def test_is_sub_class_order(self):
+    self.assertTrue(native_type_compatibility._safe_issubclass(
+        parent=typing.Dict[bytes, int],
+        derived=typing.Dict))
+    self.assertTrue(native_type_compatibility._safe_issubclass(
+        parent=typing.List[bytes],
+        derived=typing.List))
+    self.assertTrue(native_type_compatibility._safe_issubclass(
+        parent=typing.Set[int],
+        derived=typing.Set))
+    self.assertTrue(native_type_compatibility._safe_issubclass(
+        parent=typing.Tuple[int],
+        derived=typing.Tuple))
+
+  @unittest.skipIf(sys.version_info.major != '3',
```

Comment:
It is  discouraged to compare to exact "3" version, as one day there may be Python 4, see also: https://docs.python.org/3/howto/pyporting.html#use-feature-detection-instead-of-version-detection

```if sys.version_info.major < 3 will be better```